cmd/profile/delete.go

```diff
@@ -33,6 +37,14 @@ func deleteProfileCommand() *cobra.Command {
 }
 
 func deleteProfile(config *restclient.Config, ns string, profileName string) error {
-	kubeClient := kubernetes.NewForConfigOrDie(config)
-	return profile.Delete(kubeClient, ns, profileName)
+	ctrl, err := controller.NewClient(config)
+	errors.CheckError(err)
+	ctx := context.Background()
+	var prof v1.Profile
+	err = ctrl.Get(ctx, types.NamespacedName{
```

Comment:
2 comments:
- Why Get() first and then Delete(), as opposed to calling Delete() directly?
- The logic should go in profile.Delete(). The caller should not know the implementation details of a profile.
- To support the case where the profile may be a legacy one, profile.Delete() could try to delete the CR first, and if it fails, try to delete the CM with the same name.

Reply:
The result of get is being used for the call to Delete.

Reply:
You have all the information you need to construct that struct. Let me know if you want to see an example.

Reply:
Done

Reply:
Validation:

Deleting a dynamic profile CR:
```shell
ubuntu@ip-172-31-28-20:~/arlon$ arlon profile list
NAME              GEN  TYPE     BUNDLES                                   REPO-URL                           REPO-PATH                  OVRDS  TAGS        DESCRIPTION
dynamic-1         2    dynamic  [guestbook-static xenial-static]          http://172.17.0.1:8188/myrepo.git  profiles/dynamic-1         0      [examples]  dynamic test 1
dynamic-2-calico  2    dynamic  [calico guestbook-dynamic xenial-static]  http://172.17.0.1:8188/myrepo.git  profiles/dynamic-2-calico  0      [examples]  dynamic test 2
static-1          2    static   [guestbook-static xenial-static]          (N/A)                              (N/A)                      0      [examples]  static profile 1
ubuntu@ip-172-31-28-20:~/arlon$ arlon profile delete dynamic-1
ubuntu@ip-172-31-28-20:~/arlon$ arlon profile list
NAME              GEN  TYPE     BUNDLES                                   REPO-URL                           REPO-PATH                  OVRDS  TAGS        DESCRIPTION
dynamic-2-calico  2    dynamic  [calico guestbook-dynamic xenial-static]  http://172.17.0.1:8188/myrepo.git  profiles/dynamic-2-calico  0      [examples]  dynamic test 2
static-1          2    static   [guestbook-static xenial-static]          (N/A)                              (N/A)                      0      [examples]  static profile 1
```

Deleting a legacy profile ConfigMap:
```shell
# we create this fake cm to act as our legacy profile
ubuntu@ip-172-31-28-20:~/arlon$ kubectl create cm -n arlon legacy-mock --from-literal key1=value1 --from-literal key2=value2
configmap/legacy-mock created
# profile list doesn't show it. Is this expected???
ubuntu@ip-172-31-28-20:~/arlon$ arlon profile list
NAME              GEN  TYPE     BUNDLES                                   REPO-URL                           REPO-PATH                  OVRDS  TAGS        DESCRIPTION
dynamic-2-calico  2    dynamic  [calico guestbook-dynamic xenial-static]  http://172.17.0.1:8188/myrepo.git  profiles/dynamic-2-calico  0      [examples]  dynamic test 2
static-1          2    static   [guestbook-static xenial-static]          (N/A)                              (N/A)                      0      [examples]  static profile 1

# deleting works for legacy profiles
ubuntu@ip-172-31-28-20:~/arlon$ arlon profile delete legacy-mock
legacy-mock not found, assuming legacy profile
```

@bcle is this behaviour alright?

Reply:
@Rohitrajak1807 Are we validating deletion of a gen1 (legacy) profile with a mock one (configmap created manually) because there is no way to create a gen1 profile now?

Reply:
yes